### PR TITLE
ENH add option for positive simulated data in make_correlated_data

### PR DESCRIPTION
**Pull request by cassiofragadantas**
This feature is necessary for the NNLS benchmark (see benchopt/benchmark_nnls#12)

### CODE REVIEW

**Comment by mathurinm** [comment]
I don't know if it's worth adding one parameter. Do you envision usecases apart from NNLS ? Otherwise this can be handled directly in `simulated.py` in the NNLS benchmark (use the function to generate X, w, then cast to positive and generate y manually) 

My 2 cents. It seems @agramfort disagrees here : https://github.com/benchopt/benchmark_nnls/pull/12#issuecomment-1181757270

**Comment by cassiofragadantas** [comment]
I'm ok with both solutions. As you prefer.

File: benchopt/datasets/simulated.py [hunk 1]
```diff
@@ -7,7 +7,8 @@
 
 def make_correlated_data(
         n_samples=100, n_features=50, n_tasks=1, rho=0.6, snr=3,
-        w_true=None, density=0.2, X_density=1, random_state=None):
+        w_true=None, density=0.2, X_density=1, random_state=None,
+        pos_data=False):
```
Comment (agramfort) [inline benchopt/datasets/simulated.py:11]:
```suggestion
        positive=False):
```

File: benchopt/datasets/simulated.py [hunk 2]
```diff
@@ -51,6 +52,8 @@ def make_correlated_data(
         make the randomness deterministic.
     X_density: float in ]0, 1]
         Proportion of elements of X which are non-zero.
+    pos_data: boolean
```
Comment (agramfort) [inline benchopt/datasets/simulated.py:55]:
```suggestion
    positive: bool
```

File: benchopt/datasets/simulated.py [hunk 3]
```diff
@@ -102,6 +105,10 @@ def make_correlated_data(
         if w_true.ndim == 1:
             w_true = w_true[:, None]
 
+    if pos_data:
+        X = abs(X)
+        w_true = abs(w_true)
```
Comment (agramfort) [inline benchopt/datasets/simulated.py:110]:
```suggestion
    if positive:
        X = np.abs(X)
        w_true = np.abs(w_true)
```

**Review by agramfort — COMMENTED** [review]
also can you add or update the test?

**Comment by cassiofragadantas** [comment]
will do

**Comment by agramfort** [comment]
I would not do y=abs(y) just to keep things simpler

**Comment by mathurinm** [comment]
I am -1 on merging this one, IMO it's only needed in the NNLS benchmark and thus should be implemented over there rather than in the main codebase.

@tomMoral if you agree we can close

**Comment by tomMoral** [comment]
Yes I agree this is very specific and having too many parameters make it harder to use/maintain so let's not add this as one can simply call `make_correlated`, take abs value and regenerate `Y`.

**Comment by tomMoral** [comment]
Thanks a lot @cassiofragadantas anyway, sorry to have you made the code for not using it in the end.

**Comment by cassiofragadantas** [comment]
No problem ;) I will make the modifications directly on the NNLS ` simulated.py`  as suggested by @mathurinm

**Pull request closed without merging**